tests/integration/executors_batching/test_encoder_executors_batching.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+
+import numpy as np
+
+from jina.executors.encoders import BaseEncoder
+from jina.executors.decorators import batching, single, as_ndarray
+from jina import Document
+from jina.types.sets import DocumentSet
+
+EMBED_SIZE = 10
+
+
+class DummyEncoderTextBatching(BaseEncoder):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @as_ndarray
+    @batching(batch_size=3)
+    def encode(self, data, *args, **kwargs):
+        assert isinstance(data, np.ndarray)
+        assert isinstance(data[0], str)
+        assert data.shape[0] == 3
+        return np.random.random((data.shape[0], EMBED_SIZE))
+
+
+class DummyEncoderTextSingle(BaseEncoder):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @as_ndarray
+    @single
+    def encode(self, data, *args, **kwargs):
+        assert isinstance(data, str)
+        return np.random.random(EMBED_SIZE)
+
+
+@pytest.mark.parametrize('encoder', [DummyEncoderTextSingle(), DummyEncoderTextBatching()])
+def test_batching_encode_text(encoder):
+    docs = DocumentSet([Document(text=f'text-{i}') for i in range(15)])
```

Comment:
I just ran this with empty content/text and the test still passes. Is that a corner case we want?

Reply:
Seems that `.text` is empty string by default when you don't assign anything to the `Document`. I think it should be `None`, right?

Reply:
could be, but that is another issue in my opinion?